addons/auth_totp_mail/controllers/home.py

```diff
@@ -16,7 +16,7 @@ def web_totp(self, redirect=None, **kwargs):
             # In case the response from the super is a redirection
             # or the user has another TOTP method, we return the response from the call to super.
             return response
-        assert request.session.pre_uid and not request.session.uid, \
+        assert request.session['pre_uid'] and not request.session.uid, \
```

Comment:
Unrelated to your change. I would change it for an actual error, `assert` can be bypassed when running python with `-O` and I don't think this one should be bypassed.